Zend/zend_vm_execute.h

```diff
@@ -1779,6 +1772,27 @@ static ZEND_OPCODE_HANDLER_RET ZEND_FASTCALL ZEND_HANDLE_EXCEPTION_SPEC_HANDLER(
 
 	cleanup_unfinished_calls(execute_data, throw_op_num);
 
+	if (throw_op->result_type & (IS_VAR | IS_TMP_VAR)) {
+		switch (throw_op->opcode) {
+			case ZEND_ADD_ARRAY_ELEMENT:
+			case ZEND_ROPE_ADD:
+				break; /* exception while building structures, live range handling will free those */
+
+			case ZEND_FETCH_CLASS:
+			case ZEND_DECLARE_CLASS:
+			case ZEND_DECLARE_INHERITED_CLASS:
+			case ZEND_DECLARE_ANON_CLASS:
+			case ZEND_DECLARE_ANON_INHERITED_CLASS:
+				break; /* return value is zend_class_entry pointer */
+
+			case ZEND_JMP_SET:
+				break; /* takes care of handling itself */
```

Comment:
In the other branch we anyway need an `EG(exception)` check no matter what, due to the `ZEND_VM_JMP()` (otherwise we'd end up in the next opcode with EG(exception) being set - which is bad).

Also, i_zend_is_true() can also throw in case of object (e.g. in an error handler when a `Object of class %s could not be converted to boolean` error is thrown) … thus it's not the only case.